src/process/enumeration.rs

```diff
@@ -0,0 +1,85 @@
+use super::{Enumeration, EnumerationMember};
+use log::warn;
+
+use crate::api_map::APIMap;
+
+/// Convert a snake_case or SCREAMING_SNAKE_CASE string to PascalCase
+fn convert_casing(input: &String) -> String {
+    let mut result = String::new();
+    result.reserve(input.len()); // Pre-allocate memory to avoid re-allocations
+    let mut capitalize_next = true;
+
+    for c in input.chars() {
+        if c == '_' {
+            capitalize_next = true;
+        } else if capitalize_next {
+            result.push_str(&c.to_uppercase().to_string());
+            capitalize_next = false;
+        } else {
+            match c.to_lowercase().next() {
+                Some(c) => result.push(c),
+                None => result.push(c),
+            }
+        }
+    }
+
+    result
+}
+
+/// Remove the common string from the beginning of the identifier
+pub fn remove_common_string(input: &String, identifier: &String) -> String {
```

Comment:
`&String` requires the caller to have a heap-allocated string, whereas `&str` does not
You can probably say "but doesn't that require usage of `.as_str()`?", and, well, no, the Rust folks are smart, there is an implicit type cast happening, so `&str` is basically equivalent to `&String`, but allows more freedom for the caller!